src/storage.zig

```diff
@@ -299,7 +300,8 @@ pub const Storage = struct {
         // - Another replica requested a block, but we are lagging far behind, and the block address
         //   requested is beyond the end of our data file.
         if (bytes_read == 0) {
-            read.callback(read);
+            @memset(read.buffer, 0);
+            self.start_read(read, read.buffer.len);
```

Comment:
Going through `start_read` rather then directly invoking callback to have "a single return" from the async loop.